core/src/main/java/nl/nn/adapterframework/util/MessageUtils.java

```diff
@@ -87,16 +72,52 @@ public static String computeContentType(Message message) {
 			return null;
 		}
 
-		String mimeType = (String)message.getContext().get(MessageContext.METADATA_MIMETYPE);
-		if(StringUtils.isEmpty(mimeType)) {
+		MimeType mimeType = (MimeType)message.getContext().get(MessageContext.METADATA_MIMETYPE);
+		if(mimeType == null) {
 			return null;
 		}
 
-		StringBuilder contentType = new StringBuilder(mimeType);
+		StringBuilder contentType = new StringBuilder();
+		contentType.append(mimeType.getType());
+		contentType.append('/');
+		contentType.append(mimeType.getSubtype());
+
 		if(!message.isBinary()) {
 			contentType.append(";charset=");
 			contentType.append(message.getCharset());
 		}
```

Comment:
Hoewel dit oude code is, toch: Dit zou toch niet goed moeten gaan: Als de message niet binary is, heeft hij geen charset.
Als hij dat wel heeft, is het charsetveld misbruikt.

Reply:
Poeh, lastig. Zeggen wij, text heeft een charset, of binaire data heeft een charset?

Reply:
Tekst heeft alleen maar een charset als het bytes zijn